tensorflow/contrib/seq2seq/python/ops/attention_wrapper.py

```diff
@@ -1124,7 +1125,14 @@ def __init__(self,
         (default), use the context as attention at each time step. Otherwise,
         feed the context and cell output into the attention layer to generate
         attention at each time step. If attention_mechanism is a list,
-        attention_layer_size must be a list of the same length.
+        attention_layer_size must be a list of the same length. If
+        attention_layer is set, this must be None.
+      attention_layer: A list of `tf.layers.Layer` instances or a
+        single `tf.layers.Layer` instance taking the context and cell output as
+        inputs to generate attention at each time step. If None (default), use
+        the context as attention at each time step. If attention_mechanism is a
+        list, attention_layer must be a list of the same length. If
```

Comment:
What happens if mechanisms is not a list but this is?

Reply:
`attention_mechanism` is converted to a `tuple` in the constructor when it is not a sequence. The opposite is also true.